openedx/core/djangoapps/safe_sessions/middleware.py

```diff
@@ -284,7 +286,18 @@ def verify(self, user_id):
             ).unsign_object(self.signature, serializer=signing.JSONSerializer, max_age=settings.SESSION_COOKIE_AGE)
 
             if unsigned_data == self._compute_digest(user_id):
+                set_custom_attribute('SafeCookieData.verify.alg', settings.DEFAULT_HASHING_ALGORITHM)
                 return True
+            # Fallback when DEFAULT_HASHING_ALGORITHM is upgraded to 256
+            if settings.DEFAULT_HASHING_ALGORITHM != 'sha1':
+                # if not verified from DEFAULT_HASHING_ALGORITHM, try to verify from sha1
+                unsigned_data = signing.TimestampSigner(
+                    salt=self.key_salt, algorithm="sha1"
+                ).unsign_object(self.signature, serializer=signing.JSONSerializer, max_age=settings.SESSION_COOKIE_AGE)
+
+                if unsigned_data == self._compute_digest(user_id):
+                    set_custom_attribute('SafeCookieData.verify.alg', "sha1")
```

Comment:
Nit: If we switch the above value to `default`, it doesn't really matter which we use here. So, just something to consider to make it clear what line is being executed.
Again, I also changed `verify` to `verified`.
```suggestion
                    set_custom_attribute('SafeCookieData.verified.alg', "fallback")
```